core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,116 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import hudson.model.TopLevelItemDescriptor;
+import jenkins.model.Messages;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ *
+ * @since TODO
+ */
+public abstract class ItemCategory implements ExtensionPoint {
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Helpful to set the order.
+     */
+    private int weight = 1;
+
+    /**
+     * Identifier, e.g. "standaloneprojects", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * A human readable name.
+     *
+     * @return the display name
+     */
+    public abstract String getDisplayName();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
+
+    protected void setWeight(int weight) {
```

Comment:
:ant: why is this even in the API, if it is being set automatically? Could be `private` (or just inlined).